server/resp/src/main/java/org/infinispan/server/resp/tx/RespTransactionHandler.java

```diff
@@ -51,9 +55,9 @@ protected CompletionStage<RespRequestHandler> actualHandleRequest(ChannelHandler
       // Subscribe commands discard the queue and enter into pub-sub. See: https://github.com/redis/redis/pull/9928
       // Doing specific checks here instead of implementing on the commands, so we can update this later, if necessary.
       if (command instanceof SUBSCRIBE || command instanceof PSUBSCRIBE) {
-         dropTransaction();
+         CompletionStage<?> drop = dropTransaction(ctx);
          SubscriberHandler subscriberHandler = new SubscriberHandler(respServer(), respServer().newHandler());
-         return subscriberHandler.handleRequest(ctx, command, arguments);
+         return drop.thenCompose(ignore -> subscriberHandler.handleRequest(ctx, command, arguments));
```

Comment:
The consistency level is the same if we wait for dropTransactions to complete or not, so I would say lets not wait for it here. There is still a window where after the listeners are removed and the requests are processed that an entry may be modified.

I guess you could argue the overall window is smaller if we wait for removing all listeners as it will make the operation slower, but I don't think it is a good reason to slow this command down.